File: JavaScript/프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    // 배열의 길이가 짝수라면 홀수인덱스에 v + n
    // 배열의 길이가 홀수라면 짝수인덱스에 v + n
    return arr.map((v, i) => !(arr.length % 2)? i%2? v+n:v : !(i%2)? v+n:v);
}

File: Java/프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
import java.util.stream.*;

class Solution {
    public int[] solution(int[] arr, int n) {
        
        int[] answer = {};
        boolean isEven = arr.length % 2 == 0? true : false;
        return IntStream.rangeClosed(0, arr.length - 1).map(i -> {
            if(isEven) {
                if(i % 2 == 0) return arr[i];
                else return arr[i] + n;
            } else {
                if(i % 2 == 0) return arr[i] + n;
                else return arr[i];
            }
        }).toArray();
        
    }
}